src/main/java/xyz/wagyourtail/jsmacros/client/api/classes/PlayerInput.java

```diff
@@ -196,19 +194,17 @@ public static PlayerInput fromJson(String json) {
      *
      * @param input Map to be converted
      * @return The map converted into a {@code PlayerInput}
-     * @see #fromCsv(String, String)
+     * @see #fromCsv(String)
      * @since 1.4.0
      */
-    private static PlayerInput fromMap(Map<String, String> input) {
-        return new PlayerInput(
-                input.get("movementforward") != null ? Float.parseFloat(input.get("movementforward")) : input.get("w").equals(input.get("s")) ? 0.0F : (Boolean.parseBoolean(input.get("w")) ? 1.0F : -1.0F),
-                input.get("movementsideways") != null ? Float.parseFloat(input.get("movementsideways")) : input.get("a").equals(input.get("d")) ? 0.0F : (Boolean.parseBoolean(input.get("a")) ? 1.0F : -1.0F),
-                input.get("yaw") != null ? Float.parseFloat(input.get("yaw")) : 0.0f,
-                input.get("pitch") != null ? Float.parseFloat(input.get("pitch")) : 0.0f,
-                input.get("jumping") != null ? Boolean.parseBoolean(input.get("jumping")) : input.get("jump") != null && Boolean.parseBoolean(input.get("jump")),
-                input.get("sneaking") != null ? Boolean.parseBoolean(input.get("sneaking")) : input.get("sneak") != null && Boolean.parseBoolean(input.get("sneak")),
-                input.get("sprinting") != null ? Boolean.parseBoolean(input.get("sprinting")) : input.get("sprint") != null && Boolean.parseBoolean(input.get("sprint"))
-        );
+    private static PlayerInput fromMap(Map<String, String> input) throws NoSuchFieldException, IllegalAccessException {
+        PlayerInput playerInput = new PlayerInput();
+        for (Map.Entry<String, String> entry : input.entrySet()) {
+            Field field = PlayerInput.class.getDeclaredField(entry.getKey());
+            if (Modifier.isPrivate(field.getModifiers())) throw new IllegalAccessException();
```

Comment:
why reflection, :concern:

Reply:
Well it makes the code cleaner in my opinion. And my goal in the beginning was to support macros and recordings from the mod MPK, but this didn't fully work since their pitch and yaw are relativ so another thing i would have to add. Therefore I just switched completely to Reflection. Also in the toString method.

And if a user wants to read in arbitrary csvs they can always transform them in js.